pkg/applier/inventory/inventory_converter_test.go

```diff
@@ -60,21 +61,47 @@ func (f resourceGroupFactory) build(t *testing.T) *unstructured.Unstructured {
 	rg.Status = f.status
 	uObj, err := kinds.ToUnstructured(rg, core.Scheme)
 	require.NoError(t, err)
+	return rg, uObj
+}
+
+func (f resourceGroupFactory) build(t *testing.T) *unstructured.Unstructured {
+	_, uObj := f.buildBoth(t)
+	return uObj
+}
+
+func (f resourceGroupFactory) buildSpec(t *testing.T) *unstructured.Unstructured {
+	rg, uObj := f.buildBoth(t)
+	// Remove "omitempty" status fields.
+	if reflect.ValueOf(rg.Status).IsZero() {
+		unstructured.RemoveNestedField(uObj.Object, "status")
+	}
+	return uObj
+}
+
+func (f resourceGroupFactory) buildStatus(t *testing.T) *unstructured.Unstructured {
+	rg, uObj := f.buildBoth(t)
+	// Remove "omitempty" spec fields.
+	if reflect.ValueOf(rg.Spec).IsZero() {
+		unstructured.RemoveNestedField(uObj.Object, "spec")
+	}
 	return uObj
 }
 
 func TestInventoryFromUnstructured(t *testing.T) {
-	testCases := map[string]struct {
+	testCases := []struct {
+		name            string
```

Comment:
nit: I prefer map to enforce unique test case names

Normally wouldn't mention it but this seems like an unnecessary change

Reply:
Lists execute in a predictable order, which makes debugging them easier.